src/Types/BigNum.js

```diff
@@ -14,15 +14,34 @@ exports.zero = lib.BigNum.zero();
 exports.one = lib.BigNum.one();
 
 exports.bnAdd = maybe => lhs => rhs => {
+  const check_add = (a, b, limit) => {
```

Comment:
These two functions can be factored out into a single one. Provide the operation (`+` or `*`) as a function-parameter.